src/core/friendly_errors/param_validator.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * @for p5
+ * @requires core
+ */
+// import p5 from '../main';
+import dataDoc from '../../../docs/parameterData.json';
+import { z } from 'zod';
+import { fromError } from 'zod-validation-error';
+
+let schemaRegistry = new Map();
+const arrDoc = JSON.parse(JSON.stringify(dataDoc));
+
+const schemaMap = {
+  'Any': z.any(),
+  'Array': z.array(z.any()),
+  'Boolean': z.boolean(),
+  'Function': z.function(),
+  'Integer': z.number().int(),
+  'Number': z.number(),
+  'Number[]': z.array(z.number()),
+  'Object': z.object({}),
+  // Allows string for any regex
+  'RegExp': z.string(),
+  'String': z.string(),
+  'String[]': z.array(z.string())
+};
+
+function generateZodSchemasForFunc(func) {
+  const ichDot = func.lastIndexOf('.');
+  const funcName = func.slice(ichDot + 1);
+  const funcClass = func.slice(0, ichDot !== -1 ? ichDot : 0) || 'p5';
+
+  let funcInfo = arrDoc[funcClass][funcName];
+
+  let overloads = [];
+  if (funcInfo.hasOwnProperty('overloads')) {
+    overloads = funcInfo.overloads;
+  }
+
+  const createParamSchema = param => {
+    const optional = param.endsWith('?');
+    param = param.replace(/\?$/, '');
+
+    if (param.includes('|')) {
```

Comment:
We might have to do something a bit more complicated to handle cases like the `lerpPalette` one, which ends up looking like `[p5.Color, Number][]`. A quick fix could be to just add that to the schema map for now though.

I'm also noticing that we have some options objects (e.g. for `p5.Geometry.computeNormals()`) that currently end up just typed as `Object`, whereas in `docs/data.json` it has this extra `properties` info that we don't yet use:
```json
{
  "title": "param",
  "name": "options",
  "lineNumber": 11,
  "description": {
    "type": "root",
    "children": [
      {
        "type": "paragraph",
        "children": [
          {
            "type": "text",
            "value": "An optional object with configuration."
          }
        ]
      }
    ]
  },
  "type": {
    "type": "OptionalType",
    "expression": {
      "type": "NameExpression",
      "name": "Object"
    }
  },
  "properties": [
    {
      "title": "param",
      "name": "options.roundToPrecision",
      "lineNumber": 423,
      "default": "3"
    }
  ],
  "default": "{}"
}
```
That could be something we try to encode in the type eventually, e.g. instead of `Object`, it could be `{ roundToPrecision: number? }`. We can decide that's out of scope for now because that would probably necessitate a recursive parser to extract the type info (or just more structured info in `parameterData.json` rather than just a type string.)

Reply:
Thank you for bringing these edge cases up Dave! I'm adding both of these scenarios in my notes and will circle back to them later!